src/tests/ftest/cart/util/cart_utils.py

```diff
@@ -381,6 +384,53 @@ def build_cmd(self, env, host, **kwargs):
 
         return tst_cmd
 
+    def log_check_valgrind_memcheck(self):
+        """Check valgrind memcheck log files for errors."""
+
+        memcheck_errors = 0
+
+        daos_test_shared_dir = os.getenv('DAOS_TEST_SHARED_DIR',
+                                         os.getenv('HOME'))
+
+        self.log.info("Parsing log path %s", daos_test_shared_dir)
+        if not os.path.exists(daos_test_shared_dir):
+            self.log.info("Path does not exist")
+            return 1
+
+        xml_filename_fmt = r"^valgrind\.\S+\.memcheck$"
+        memcheck_files = list(filter(lambda x: re.match(xml_filename_fmt, x),
+                                os.listdir(daos_test_shared_dir)))
+
+        for filename in memcheck_files:
```

Comment:
One thing we do in NLT is to change the filenames used in the xml file which doesn't appear to be happening here, the xml contains "/home/jenkins/build/<job>/<job-id>/src/cart/crt_init.c" which is technically correct, but the convert_xml function changes this to "src/cart/crt_init.c" which is more helpful and allows Jenkins to render the xml files correctly and show the source.

Reply:
Unfortunately, we only have object and function names in the memcheck output (no filename or line numbers), e.g.,

```xml
  ...
  <frame>
    <ip>0x54CE0EF</ip>
    <obj>/usr/lib64/libcart.so.4.9.0</obj>
    <fn>crt_hg_ctx_init</fn>
  </frame>
  <frame>
    <ip>0x549A285</ip>
    <obj>/usr/lib64/libcart.so.4.9.0</obj>
    <fn>crt_context_provider_create</fn>
  </frame>
  <frame>
    <ip>0x4077A1</ip>
    <obj>/usr/lib/daos/TESTING/tests/test_proto_client</obj>
  </frame>
  <frame>
    <ip>0x401ECF</ip>
    <obj>/usr/lib/daos/TESTING/tests/test_proto_client</obj>
  </frame>
```

Reply:
That seems wrong, I assume it's because you're testing with a "release" build, however even they should be using -g which would give file/line numbers.

Reply:
[`--track-origins=yes`](https://valgrind.org/docs/manual/manual-core.html) valgrind option seems to resolve the problem.  Now I get something like this in the output:
```xml
...
    <frame>
      <ip>0x40236A</ip>
      <obj>/home/emallovx/daos/install/lib/daos/TESTING/tests/iv_client</obj>
      <fn>crt_init</fn>
      <dir>/home/emallovx/daos/src/include/cart</dir>
      <file>api.h</file>
      <line>72</line>
    </frame>
    <frame>
      <ip>0x40236A</ip>
      <obj>/home/emallovx/daos/install/lib/daos/TESTING/tests/iv_client</obj>
      <fn>main</fn>
      <dir>/home/emallovx/daos/src/tests/ftest/cart</dir>
      <file>iv_client.c</file>
      <line>628</line>
```